### PR TITLE
Web blog Update

### PR DESCRIPTION
**Pull request by LUS24**
Some improvements are:

Use of [Flask CLI](https://flask.palletsprojects.com/en/2.0.x/cli/#command-line-interface) to allow easier application discovery using an environmental variable FLASK_APP.
Added an improved README.md file with details about how to use the code.
Other updates:

Pinned packages to the latest versions.
Added .venv directory to .gitignore.
Added annotations to requirements.txt file.
Added new requirement python-dotenv.
Updated the database.py file to user insert_one instead of insert for compatibility with pymongo >4.0

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,3 +16,53 @@ The available endpoints are:
 - `/blogs/new`
 - `/posts/<string:blog_id>`
 - `/posts/new/<string:blog_id>`
+
+
+## HOW TO USE THIS CODE
+
+1) Create a new virtual environment: `python -m venv venv`
```
Comment (jslvtr) [inline README.md]: To stick to what we do in other parts of the course (especially the new parts), could you use `.venv` as the venv name?

File: README.md [hunk 2]
```diff
@@ -16,3 +16,53 @@ The available endpoints are:
...
+
+```bash
+source venv/Scripts/activate
+```
+
+In CMD
+
+```cmd
+venv\Scripts\activate
+```
```
Comment (jslvtr) [inline README.md]: Can we include a Mac OS / Linux version?

It's:

```
source .venv/bin/activate
```

File: README.md [hunk 3]
```diff
@@ -16,3 +16,53 @@ The available endpoints are:
...
+
+```bash
+export FLASK_APP=src.app
+```
+
+In CMD
+
+```cmd
+set FLASK_APP=src.app
+```
```
Comment (jslvtr) [inline README.md:48]: The MacOS/Linux version is the same as the bash version here.

Regarding the `.flaskenv` file, do we want to add it to this repo? Then they don't have to set the FLASK_APP value themselves

File: src/common/database.py [hunk 4]
```diff
@@ -14,7 +14,7 @@ def initialize():
 
     @staticmethod
     def insert(collection, data):
-        Database.DATABASE[collection].insert(data)
+        Database.DATABASE[collection].insert_one(data)
```
Comment (jslvtr) [inline src/common/database.py:17]: Nice surreptitious fix of a long-standing bug 👍

**Review by jslvtr — CHANGES_REQUESTED** [review]
Looks great! Just a couple things to add, MacOS info + potentially a simplification by using `.flaskenv`

File: README.md (hunk 1 above)
Comment (LUS24) [reply README.md]: Updated

File: README.md (hunk 2 above)
Comment (LUS24) [reply README.md]: Added

File: README.md (hunk 3 above)
Comment (LUS24) [reply README.md:48]: Added

**Review by jslvtr — APPROVED** [review]
Nice one!

**Pull request merged**